### PR TITLE
lightgbm 3.2.0

### PR DESCRIPTION
**Pull request by chenrui333**
[`action-homebrew-bump-formula`](https://github.com/dawidd6/action-homebrew-bump-formula)

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/lightgbm.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Lightgbm < Formula
   desc "Fast, distributed, high performance gradient boosting framework"
   homepage "https://github.com/microsoft/LightGBM"
-  url "https://github.com/microsoft/LightGBM/archive/v3.1.1.tar.gz"
-  sha256 "a1e64bff404f448681ee33fb131a6218c4388574b83df0d25f45e63268a03b44"
+  url "https://github.com/microsoft/LightGBM/archive/v3.2.0.tar.gz"
+  sha256 "13cb5ba45f6bb26537090f40564d425a37d9b9ef7dd8b5e82d7976fe2434c771"
```
Comment (StrikerRUS) [inline Formula/lightgbm.rb:5]:
```suggestion
  url "https://github.com/microsoft/LightGBM/releases/download/v3.2.0/LightGBM-3.2.0-complete_source_code_tar_gz.tar.gz"
  sha256 "a470b46cc5768f119b7ce5c79a38b8c88ca8d3d7db5d958f6c38f37e4fb74b83"
```

**Review by StrikerRUS — COMMENTED** [review]
@chenrui333 Thank you very much for updating the formula!

Reason of failing CI is in GitHub doesn't include submodules into auto-created artifacts. And it seems that they don't have plans to do this. I've just uploaded complete sources that should be built OK.

Refer to the discussion https://github.com/microsoft/LightGBM/pull/3872#issuecomment-804087682 for more details.

**Comment by SMillerDev** [comment]
Maybe we're better of with a checkout with submodules then?

**Comment by StrikerRUS** [comment]
@SMillerDev 

> Maybe we're better of with a checkout with submodules then?

If this is supported by Homebrew and not hard to implement, I think it will be more reliable than handcrafted archives.
But we going to automate the process of publishing complete source code at CI level, so there will be no time delta between new release and attaching correct `*.tar.gz` artifact.

**Comment by SMillerDev** [comment]
You just switch from an archive to specifying a tag and a revision.

File: Formula/lightgbm.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 class Lightgbm < Formula
   desc "Fast, distributed, high performance gradient boosting framework"
   homepage "https://github.com/microsoft/LightGBM"
-  url "https://github.com/microsoft/LightGBM/archive/v3.1.1.tar.gz"
-  sha256 "a1e64bff404f448681ee33fb131a6218c4388574b83df0d25f45e63268a03b44"
+  url "https://github.com/microsoft/LightGBM/releases/download/v3.2.0/LightGBM-3.2.0-complete_source_code_tar_gz.tar.gz"
+  sha256 "a470b46cc5768f119b7ce5c79a38b8c88ca8d3d7db5d958f6c38f37e4fb74b83"
```
Comment (StrikerRUS) [inline Formula/lightgbm.rb:5]:
```suggestion
  url "https://github.com/microsoft/LightGBM.git",
      tag:      "v3.2.0",
      revision: "2e54d5f518a4c7db721483347bec6676a8d20c89"
```

**Review by StrikerRUS — COMMENTED** [review]
@SMillerDev 
> You just switch from an archive to specifying a tag and a revision.

Like this one?

File: Formula/lightgbm.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/lightgbm.rb:5]: Yeah, that will work.

**Comment by carlocab** [comment]
Some of these external libs are provided by Homebrew (e.g. eigen, fmt). Can we use Homebrew dependencies for those?

**Comment by StrikerRUS** [comment]
@carlocab 

> Some of these external libs are provided by Homebrew (e.g. eigen, fmt). Can we use Homebrew dependencies for those?

To be honest, I don't think it is possible and good idea.

We run tests in LightGBM only against fixed commits of those external libs. New versions might be incompatible and broke the code. If Homebrew is able to specify exact commit hash for a dependency, it is not a problem, though.

More serious problem is that LightGBM expects all these dependencies to be located in a specific place (we don't respect system-wide installations), e.g.
https://github.com/microsoft/LightGBM/blob/4e47c9350828aca5c86ee4221cd72a9aac5e3796/include/LightGBM/utils/common.h#L33-L37

**Comment by StrikerRUS** [comment]
> compute should probably be provided by `boost`

`boost` is not needed for ordinary (currently the only available option in Homebrew) installation of LightGBM.

**Comment by SMillerDev** [comment]
> We run tests in LightGBM only against fixed commits of those external libs. New versions might be incompatible and broke the code. If Homebrew is able to specify exact commit hash for a dependency, it is not a problem, though.

That is conflicting with our requirements for formulae: https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

**Review by carlocab — DISMISSED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**